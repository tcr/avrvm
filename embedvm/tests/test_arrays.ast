     LEFT 0x7fdd8ac03ba0 @ 0000 0000:
      RIGHT 0x7fdd8ac03b10 @ 0000 0000: sym=a8u datalen=5 initdata=0102030405
    LEFT 0x7fdd8ac03d80 @ 0000 0005:
     RIGHT 0x7fdd8ac03cf0 @ 0005 0005: sym=a8s datalen=5 initdata=fffefdfcfb
   LEFT 0x7fdd8ac03f60 @ 0000 000a:
    RIGHT 0x7fdd8ac03ed0 @ 000a 000a: sym=a16 datalen=10 initdata=03e807d00bb80fa01388
  LEFT 0x7fdd8ac04a80 @ 0000 0014:
     LEFT 0x7fdd8ac04980 @ 0014 0014: op=f0
    LEFT 0x7fdd8ac049c0 @ 0014 0015:
      LEFT 0x7fdd8ac04830 @ 0015 0015:
         LEFT 0x7fdd8ac04040 @ 0015 0015: op=90
        LEFT 0x7fdd8ac04080 @ 0015 0016:
         RIGHT 0x7fdd8ac04000 @ 0016 0016: op=40
       RIGHT 0x7fdd8ac047f0 @ 0015 0017:
           LEFT 0x7fdd8ac040d0 @ 0017 0017: op=00
          LEFT 0x7fdd8ac04150 @ 0017 0018:
           RIGHT 0x7fdd8ac04110 @ 0018 0018: op=98 arg=05
         LEFT 0x7fdd8ac04190 @ 0017 001a: op=a8
        RIGHT 0x7fdd8ac047b0 @ 0017 001b: op=a6 arg=14 rel argaddr=0x7fdd8ac04340
                LEFT 0x7fdd8ac04570 @ 001d 001d: op=00
               LEFT 0x7fdd8ac045b0 @ 001d 001e: op=e3 arg=0a argaddr=0x7fdd8ac03ed0
              LEFT 0x7fdd8ac045f0 @ 001d 0020:
                 LEFT 0x7fdd8ac04490 @ 0020 0020: op=00
                LEFT 0x7fdd8ac044d0 @ 0020 0021: op=d3 arg=05 argaddr=0x7fdd8ac03cf0
               RIGHT 0x7fdd8ac04510 @ 0020 0023:
                 LEFT 0x7fdd8ac043e0 @ 0023 0023: op=00
                RIGHT 0x7fdd8ac04420 @ 0023 0024: op=c3 arg=00 argaddr=0x7fdd8ac03b10
             LEFT 0x7fdd8ac04630 @ 001d 0026: op=93
            LEFT 0x7fdd8ac04670 @ 001d 0027: op=b1
           LEFT 0x7fdd8ac046b0 @ 001d 0028: op=9d
          LEFT 0x7fdd8ac04730 @ 001d 0029:
            LEFT 0x7fdd8ac04380 @ 0029 0029:
           RIGHT 0x7fdd8ac046f0 @ 0029 0029:
              LEFT 0x7fdd8ac041e0 @ 0029 0029: op=00
             LEFT 0x7fdd8ac042a0 @ 0029 002a:
               LEFT 0x7fdd8ac04220 @ 002a 002a: op=91
              RIGHT 0x7fdd8ac04260 @ 002a 002b: op=80
            RIGHT 0x7fdd8ac042e0 @ 0029 002c: op=40
         RIGHT 0x7fdd8ac04770 @ 001d 002d: op=a0 arg=ea rel argaddr=0x7fdd8ac04190
          RIGHT 0x7fdd8ac04340 @ 002f 002f:
     RIGHT 0x7fdd8ac04940 @ 0015 002f:
        LEFT 0x7fdd8ac04880 @ 002f 002f: op=90
       LEFT 0x7fdd8ac048c0 @ 002f 0030: op=b2
      RIGHT 0x7fdd8ac04900 @ 002f 0031: op=9d
   RIGHT 0x7fdd8ac04a00 @ 0014 0032: sym=print op=9c
 LEFT 0x7fdd8ac069c0 @ 0000 0033:
    LEFT 0x7fdd8ac068c0 @ 0033 0033: op=f0
   LEFT 0x7fdd8ac06900 @ 0033 0034:
        LEFT 0x7fdd8ac04bb0 @ 0034 0034:
          LEFT 0x7fdd8ac04b30 @ 0034 0034: op=a2 arg=e0 rel argaddr=0x7fdd8ac04a00
         RIGHT 0x7fdd8ac04b70 @ 0034 0036: op=9d
       LEFT 0x7fdd8ac057e0 @ 0034 0037:
          LEFT 0x7fdd8ac04c40 @ 0037 0037: op=90
         LEFT 0x7fdd8ac04c80 @ 0037 0038:
          RIGHT 0x7fdd8ac04c00 @ 0038 0038: op=40
        RIGHT 0x7fdd8ac057a0 @ 0037 0039:
            LEFT 0x7fdd8ac04cd0 @ 0039 0039: op=00
           LEFT 0x7fdd8ac04d50 @ 0039 003a:
            RIGHT 0x7fdd8ac04d10 @ 003a 003a: op=98 arg=05
          LEFT 0x7fdd8ac04d90 @ 0039 003c: op=a8
         RIGHT 0x7fdd8ac05760 @ 0039 003d: op=a6 arg=23 rel argaddr=0x7fdd8ac04f40
             LEFT 0x7fdd8ac051e0 @ 003f 003f:
                   LEFT 0x7fdd8ac04fe0 @ 003f 003f: op=00
                  LEFT 0x7fdd8ac050e0 @ 003f 0040: op=c5
                 LEFT 0x7fdd8ac05020 @ 003f 0041: op=c3 arg=00 argaddr=0x7fdd8ac03b10
                LEFT 0x7fdd8ac05120 @ 003f 0043:
                  LEFT 0x7fdd8ac05060 @ 0043 0043: op=91
                 RIGHT 0x7fdd8ac050a0 @ 0043 0044: op=80
               LEFT 0x7fdd8ac05160 @ 003f 0045: op=c6
              RIGHT 0x7fdd8ac051a0 @ 003f 0046: op=cb arg=00 argaddr=0x7fdd8ac03b10
            LEFT 0x7fdd8ac05660 @ 003f 0048:
                  LEFT 0x7fdd8ac05240 @ 0048 0048: op=00
                 LEFT 0x7fdd8ac05560 @ 0048 0049: op=c5
                LEFT 0x7fdd8ac05280 @ 0048 004a: op=e3 arg=0a argaddr=0x7fdd8ac03ed0
               LEFT 0x7fdd8ac055a0 @ 0048 004c:
                       LEFT 0x7fdd8ac052e0 @ 004c 004c: op=00
                      LEFT 0x7fdd8ac053e0 @ 004c 004d: op=c5
                     LEFT 0x7fdd8ac05320 @ 004c 004e: op=d3 arg=05 argaddr=0x7fdd8ac03cf0
                    LEFT 0x7fdd8ac05420 @ 004c 0050:
                      LEFT 0x7fdd8ac05360 @ 0050 0050: op=91
                     RIGHT 0x7fdd8ac053a0 @ 0050 0051: op=81
                   LEFT 0x7fdd8ac05460 @ 004c 0052: op=cd
                  LEFT 0x7fdd8ac054a0 @ 004c 0053: op=c6
                 LEFT 0x7fdd8ac054e0 @ 004c 0054: op=db arg=05 argaddr=0x7fdd8ac03cf0
                RIGHT 0x7fdd8ac05520 @ 004c 0056: op=81
              LEFT 0x7fdd8ac055e0 @ 0048 0057: op=c6
             RIGHT 0x7fdd8ac05620 @ 0048 0058: op=eb arg=0a argaddr=0x7fdd8ac03ed0
           LEFT 0x7fdd8ac056e0 @ 003f 005a:
             LEFT 0x7fdd8ac04f80 @ 005a 005a:
            RIGHT 0x7fdd8ac056a0 @ 005a 005a:
               LEFT 0x7fdd8ac04de0 @ 005a 005a: op=00
              LEFT 0x7fdd8ac04ea0 @ 005a 005b:
                LEFT 0x7fdd8ac04e20 @ 005b 005b: op=91
               RIGHT 0x7fdd8ac04e60 @ 005b 005c: op=80
             RIGHT 0x7fdd8ac04ee0 @ 005a 005d: op=40
          RIGHT 0x7fdd8ac05720 @ 003f 005e: op=a0 arg=db rel argaddr=0x7fdd8ac04d90
           RIGHT 0x7fdd8ac04f40 @ 0060 0060:
      LEFT 0x7fdd8ac058c0 @ 0034 0060:
        LEFT 0x7fdd8ac05840 @ 0060 0060: op=a2 arg=b4 rel argaddr=0x7fdd8ac04a00
       RIGHT 0x7fdd8ac05880 @ 0060 0062: op=9d
     LEFT 0x7fdd8ac067a0 @ 0034 0063:
        LEFT 0x7fdd8ac05950 @ 0063 0063: op=90
       LEFT 0x7fdd8ac05990 @ 0063 0064:
        RIGHT 0x7fdd8ac05910 @ 0064 0064: op=40
      RIGHT 0x7fdd8ac06760 @ 0063 0065:
          LEFT 0x7fdd8ac059e0 @ 0065 0065: op=00
         LEFT 0x7fdd8ac05a60 @ 0065 0066:
          RIGHT 0x7fdd8ac05a20 @ 0066 0066: op=98 arg=05
        LEFT 0x7fdd8ac05aa0 @ 0065 0068: op=a8
       RIGHT 0x7fdd8ac06720 @ 0065 0069: op=a6 arg=27 rel argaddr=0x7fdd8ac05c50
            LEFT 0x7fdd8ac05ec0 @ 006b 006b:
                LEFT 0x7fdd8ac05d70 @ 006b 006b: op=98 arg=c8
               LEFT 0x7fdd8ac05e00 @ 006b 006d:
                RIGHT 0x7fdd8ac05dc0 @ 006d 006d: op=00
              LEFT 0x7fdd8ac05e40 @ 006b 006e: op=80
             RIGHT 0x7fdd8ac05e80 @ 006b 006f:
               LEFT 0x7fdd8ac05cf0 @ 006f 006f: op=00
              RIGHT 0x7fdd8ac05d30 @ 006f 0070: op=cb arg=00 argaddr=0x7fdd8ac03b10
           LEFT 0x7fdd8ac06330 @ 006b 0072:
               LEFT 0x7fdd8ac05fa0 @ 0072 0072: op=98 arg=64
              LEFT 0x7fdd8ac06270 @ 0072 0074:
                  LEFT 0x7fdd8ac05ff0 @ 0074 0074: op=00
                 LEFT 0x7fdd8ac06070 @ 0074 0075:
                  RIGHT 0x7fdd8ac06030 @ 0075 0075: op=91
                LEFT 0x7fdd8ac060b0 @ 0074 0076: op=87
               RIGHT 0x7fdd8ac06230 @ 0074 0077: op=a6 arg=05 rel argaddr=0x7fdd8ac061b0
                 LEFT 0x7fdd8ac060f0 @ 0079 0079: op=97
                RIGHT 0x7fdd8ac061f0 @ 0079 007a: op=a0 arg=03 rel argaddr=0x7fdd8ac06170
                  LEFT 0x7fdd8ac06130 @ 007c 007c: op=91
                 RIGHT 0x7fdd8ac061b0 @ 007c 007d:
                  RIGHT 0x7fdd8ac06170 @ 007d 007d:
             LEFT 0x7fdd8ac062b0 @ 0072 007d: op=82
            RIGHT 0x7fdd8ac062f0 @ 0072 007e:
              LEFT 0x7fdd8ac05f20 @ 007e 007e: op=00
             RIGHT 0x7fdd8ac05f60 @ 007e 007f: op=db arg=05 argaddr=0x7fdd8ac03cf0
          LEFT 0x7fdd8ac06620 @ 006b 0081:
              LEFT 0x7fdd8ac06410 @ 0081 0081: op=98 arg=6f
             LEFT 0x7fdd8ac06560 @ 0081 0083:
                LEFT 0x7fdd8ac06460 @ 0083 0083: op=00
               LEFT 0x7fdd8ac064e0 @ 0083 0084:
                RIGHT 0x7fdd8ac064a0 @ 0084 0084: op=91
              RIGHT 0x7fdd8ac06520 @ 0083 0085: op=80
            LEFT 0x7fdd8ac065a0 @ 0081 0086: op=82
           RIGHT 0x7fdd8ac065e0 @ 0081 0087:
             LEFT 0x7fdd8ac06390 @ 0087 0087: op=00
            RIGHT 0x7fdd8ac063d0 @ 0087 0088: op=eb arg=0a argaddr=0x7fdd8ac03ed0
         LEFT 0x7fdd8ac066a0 @ 006b 008a:
           LEFT 0x7fdd8ac05c90 @ 008a 008a:
          RIGHT 0x7fdd8ac06660 @ 008a 008a:
             LEFT 0x7fdd8ac05af0 @ 008a 008a: op=00
            LEFT 0x7fdd8ac05bb0 @ 008a 008b:
              LEFT 0x7fdd8ac05b30 @ 008b 008b: op=91
             RIGHT 0x7fdd8ac05b70 @ 008b 008c: op=80
           RIGHT 0x7fdd8ac05bf0 @ 008a 008d: op=40
        RIGHT 0x7fdd8ac066e0 @ 006b 008e: op=a0 arg=d7 rel argaddr=0x7fdd8ac05aa0
         RIGHT 0x7fdd8ac05c50 @ 0090 0090:
    RIGHT 0x7fdd8ac06880 @ 0034 0090:
      LEFT 0x7fdd8ac06800 @ 0090 0090: op=a2 arg=84 rel argaddr=0x7fdd8ac04a00
     RIGHT 0x7fdd8ac06840 @ 0090 0092: op=9d
  RIGHT 0x7fdd8ac06940 @ 0033 0093: sym=main op=9c
ROOT 0x7fdd8ac06a50 @ 0000 0094:
 RIGHT 0x7fdd8ac06a00 @ 0094 0094: sym=_end
