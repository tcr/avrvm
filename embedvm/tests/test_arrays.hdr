#define EMBEDVM_SYM__end 0x0094
#define EMBEDVM_SYM_main 0x0033
#define EMBEDVM_SYM_print 0x0014
#define EMBEDVM_SYM_a16 0x000a
#define EMBEDVM_SYM_a8s 0x0005
#define EMBEDVM_SYM_a8u 0x0000
#define EMBEDVM_SECT_SRAM_BEGIN 0x0000
#define EMBEDVM_SECT_SRAM_END 0xffff
#define EMBEDVM_SECT_SRAM_DATA 1,2,3,4,5,255,254,253,252,251,3,232,7,208,11,184,15,160,19,136,240,144,64,0,152,5,168,166,20,0,227,10,0,211,5,0,195,0,147,177,157,0,145,128,64,160,234,144,178,157,156,240,162,224,157,144,64,0,152,5,168,166,35,0,197,195,0,145,128,198,203,0,0,197,227,10,0,197,211,5,145,129,205,198,219,5,129,198,235,10,0,145,128,64,160,219,162,180,157,144,64,0,152,5,168,166,39,152,200,0,128,0,203,0,152,100,0,145,135,166,5,151,160,3,145,130,0,219,5,152,111,0,145,128,130,0,235,10,0,145,128,64,160,215,162,132,157,156
