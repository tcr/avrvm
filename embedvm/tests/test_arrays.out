Called user function 2 with 0 args:
Called user function 2 with 0 args:
Called user function 2 with 0 args:
Main function returned => Terminating.
